### PR TITLE
aarch64: remove x86 specific code

### PR DESCRIPTION
**Pull request by shrik3**
Also:
- guard unimplemented with todo!() macro
- remove SysArchPrctl for aarch64

### CODE REVIEW

**Comment by chl337** [comment]
@shrik3 Interestingly, I am doing the same thing! I suspect some of the strange behavior I am facing relates to some call to an empty implementation.

File: qlib/kernel/asm/aarch64.rs [hunk 1]
```diff
@@ -186,6 +173,7 @@ pub fn GetCurrentUserSp() -> u64 {
 
 #[inline]
 pub fn Clflush(addr: u64) {
```
Comment (chl337) [inline qlib/kernel/asm/aarch64.rs:175]: @shrik3 This is obsolete, as it is not used anywhere.

File: qlib/kernel/asm/aarch64.rs [hunk 2]
```diff
@@ -72,6 +58,7 @@ pub fn GetVcpuId() -> usize {
 
 #[inline]
 pub fn Hlt() {
```
Comment (chl337) [inline qlib/kernel/asm/aarch64.rs:60]: @shrik3 This method is obsolete, as it is not used anywhere.

**Comment by shrik3** [comment]
> @shrik3 Interestingly, I am doing the same thing! I suspect some of the strange behavior I am facing relates to some call to an empty implementation.

afaik "calling an empty implementation" would only apply to 
1. FPU init
2. getting FS/GS.   Also the tls should not be causing issues ofter #1092 

Anyways I don't think these would cause problem. (are there other such empty implementation that I missed?)

**Review by abel-von — APPROVED** [review]

**Pull request merged**